File: quicksort.rb
# http://rosettacode.org/wiki/Sorting_algorithms/Quicksort

def quicksort(array)
  return array if array.length <= 1
  lesser = []; equiv = []; greater = []
  pivot = array[array.length/2]
  array.each do |element|
    if element < pivot
      lesser << element
    elsif element > pivot
      greater << element
    else element == pivot
      equiv << element     
    end
  end   
  quicksort(lesser) + equiv + quicksort(greater) 
end    
        
test_array = [4, 65, 2, -31, 0, 99, 2, 83, 782, 1]
p quicksort(test_array)

File: quicksort.swift
// http://rosettacode.org/wiki/Sorting_algorithms/Quicksort

func quicksort(array:[Int]) -> [Int] {
  if array.count <= 1 {
    return array
  }
  var lesser:[Int] = []
  var greater:[Int] = []
  var equiv:[Int] = []
  let pivot = array[array.count/2]
  for element in array {
    if element < pivot {
      lesser.append(element)
    } else if element > pivot {
      greater.append(element)
    } else {
      equiv.append(element)
    }
  }
  return quicksort(lesser) + equiv + quicksort(greater)
}

let testArray:[Int] = [4,65,2,-31,0,99,2,83,782,1]
println(quicksort(testArray))
